### PR TITLE
ES module

### PR DESCRIPTION
**Pull request by 0x33467**
@targos @maasencioh

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 language: node_js
 node_js:
-  - stable
-  - v6
-  - v4
+  - node
```
Comment (targos) [inline .travis.yml]: 8

File: src/TreeNode.js [hunk 2]
```diff
@@ -203,31 +209,6 @@ class TreeNode {
 
     /**
      * @private
-     * Save the current node and their children.
-     * @return {object}
-     */
-    toJSON() {
```
Comment (targos) [inline src/TreeNode.js:209]: this is not needed anymore?

File: package.json [hunk 3]
```diff
@@ -35,16 +36,18 @@
...
-    "mocha": "^2.5.3",
-    "mocha-better-spec-reporter": "^3.0.2",
-    "should": "^10.0.0"
+    "npm-run-all": "^4.0.2",
+    "rollup": "^0.43.0",
+    "jest": "^20.0.4"
   },
   "dependencies": {
+    "ml-array-mean": "^1.0.0",
     "ml-dataset-iris": "^1.0.0",
```
Comment (targos) [inline package.json]: is that really a dependency? I guess it should be a devDependency

File: src/DecisionTreeClassifier.js [hunk 4]
```diff
@@ -1,9 +1,14 @@
-'use strict';
+import {Matrix as Matrix} from 'ml-matrix';
```
Comment (targos) [inline src/DecisionTreeClassifier.js]: you should be able to just `import Matrix`

File: src/TreeNode.js [hunk 5]
```diff
@@ -1,18 +1,17 @@
-'use strict';
+import {Matrix as Matrix} from 'ml-matrix';
```
Comment (targos) [inline src/TreeNode.js]: idem

File: src/DecisionTreeClassifier.js [hunk 6]
```diff
@@ -21,12 +26,7 @@ class DecisionTreeClassifier {
             this.root = new Tree(model.options);
             this.root.setNodeParameters(model.root);
         } else {
-            if (options === undefined) options = {};
-            if (options.gainFunction === undefined) options.gainFunction = 'gini';
-            if (options.splitFunction === undefined) options.splitFunction = 'mean';
-            if (options.minNumSamples === undefined) options.minNumSamples = 3;
-            if (options.maxDepth === undefined) options.maxDepth = Infinity;
-
+            options = Object.assign({}, defaultOptions, options);
```
Comment (maasencioh) [inline src/DecisionTreeClassifier.js:29]: why not assign directly to `this.options`?

File: src/TreeNode.js [hunk 7]
```diff
@@ -21,14 +20,21 @@ class TreeNode {
      * @constructor
      */
     constructor(options) {
-
+        // variables for the node
         this.left = undefined;
```
Comment (maasencioh) [inline src/TreeNode.js:24]: Why to create undefined variables?

File: src/TreeNode.js [hunk 8]
```diff
@@ -21,14 +20,21 @@ class TreeNode {
...
         this.right = undefined;
         this.distribution = undefined;
         this.splitValue = undefined;
         this.splitColumn = undefined;
         this.gain = undefined;
+
+        // options parameters
         this.options = options;
```
Comment (maasencioh) [inline src/TreeNode.js:32]: if all the options are the listed bellow, why keep this object?

**Review by targos — APPROVED** [review]
let's merge this and continue iterating on master. the diff is becoming big

**Pull request merged**